web-app/js/portal/filter/Filter.js

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2015 IMOS
+ *
+ * The AODN/IMOS Portal is distributed under the terms of the GNU General Public License
+ *
+ */
+
+Ext.namespace('Portal.filter');
+
+Portal.filter.Filter = Ext.extend(Object, {
+
+    constructor: function(filterDetail, layer) {
+
+        this.layer = layer;
+        this.filterName = filterDetail.name;
```

Comment:
What is filterName - would `id` be more descriptive?  Also, can we drop `filter` from this (just have `name`) since it's obvious it's a filter name from the context.